src/snarky.d.ts

```diff
@@ -51,489 +53,806 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * ```ts
+   * const x = Field(6);
+   * const quotient = x.div(Field(3));
+   * 
+   * quotient.assertEquals(Field(2));
+   * ```
+   * 
+   * **Warning**: This is a modular division in the pasta field. You can think it as the reverse operation of modular multiplication.
+   * 
+   * @example
+   * ```ts
+   * const x = Field(2);
+   * const y = Field(5);
+   * 
+   * const quotient = x.div(y);
+   * 
+   * // If you try to print quotient - `console.log(quotient.toBigInt())` - you will realize that it prints a very big integer because this is a modular inverse.
+   * // You can use the reverse operation of division (multiplication) to prove the quotient is calculated correctly.
+   * 
+   * quotient.mul(y).assertEquals(x);
+   * ```
+   * 
+   * @param value - a "field-like" value to divide with the {@link Field}.
+   * 
+   * @return A {@link Field} element equivalent to the modular division of the two value.
    */
-  div(y: Field | number | string | boolean): Field;
+  div(value: Field | number | string | boolean): Field;
 
   /**
-   * Squares this {@link Field} element.
+   * Square this {@link Field} element.
    *
-   * ```typescript
-   * const x2 = x.square();
-   * x2.assertEquals(x.mul(x));
+   * @example
+   * ```ts
+   * const someField = Field(7);
+   * const square = someField.square();
+   * 
+   * square.assertEquals(someField.mul(someField)); // This statement is always true regardless of the value of `someField`
    * ```
+   * 
+   * ** Warning: This is a modular multiplication. Please see `mul()` function for more details.
+   * 
+   * @return A {@link Field} element equivalent to the multiplication of the {@link Field} element with itself.
    */
   square(): Field;
 
   /**
-   * Square roots this {@link Field} element.
+   * Take the square root of this {@link Field} element.
    *
-   * ```typescript
-   * x.square().sqrt().assertEquals(x);
+   * @example
+   * ```ts
+   * const someField = Field(42);
+   * const squareRoot = someField.sqrt();
+   * 
+   * squareRoot.mul(squareRoot).assertEquals(someField); // This statement is always true regardless of the value of `someField`
    * ```
+   * 
+   * **Warning**: This is a modular square root. Please see `div()` function for more details.
+   * 
+   * @return A {@link Field} element equivalent to the square root of the {@link Field} element.
    */
   sqrt(): Field;
 
   /**
-   * Serialize the {@link Field} to a string, e.g. for printing.
-   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
+   * Serialize the {@link Field} to a string, e.g. for printing. If you try to print a {@link Field} without this function it will directly stringify the Field object, resulting in an unreadable output.
+   * 
+   * **Warning**: This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the {@link Field}. Please use it only during debugging.
```

Comment:
apply same suggested changes as in https://github.com/o1-labs/snarkyjs/pull/917/files#diff-c986331433c1c055aa5d451d0b789e5f5e9aee97f754e6f1550fb09b5acc91b4R323

Reply:
the goal is to provide a clear subject for "it" (in this case, it is the operation)